scripts/check_frontend_coverage.py

```diff
@@ -0,0 +1,378 @@
+
+        match = re.search('LF:(\d+)', relevant_lines[1])
+        if match is None:
+            raise Exception(
+                'It wasn\'t possible to get the total lines of {} file.'
+                'It\'s not possible to diff the test coverage correctly.'
+                .format(file_name))
+        self.total_lines = int(match.group(1))
+
+        match = re.search('LH:(\d+)', relevant_lines[2])
+        if match is None:
+            raise Exception(
+                'It wasn\'t possible to get the covered lines of {} file.'
+                'It\'s not possible to diff the test coverage correctly.'
+                .format(file_name))
+        self.covered_lines = int(match.group(1))
+
+    def get_file_name(self):
+        return self.file_name
+
+    def get_total_lines(self):
+        return self.total_lines
+
+    def get_covered_lines(self):
+        return self.covered_lines
+
+
+def run_frontend_tests_script():
+    """Run the frontend tests script using subprocess. If any test fails then
+    the coverage check won't happen.
+    """
+    try:
+        subprocess.check_call([
+            'python', '-m', 'scripts.run_frontend_tests'])
+    except subprocess.CalledProcessError:
+        sys.exit(1)
+
+
+def get_stanzas_from_lcov_file():
+    """Get all stanzas from a lcov file. The lcov file gather all the frontend
+    files that has tests and each one has the following structure:
+    TN: test name
+    SF: file path
+    FNF: total functions
+    FNH: functions covered
+    LF: total lines
+    LH: lines covered
+    BRF: total branches
+    BRH: branches covered
+    end_of_record
+
+    Returns:
+        list(LcovStanzaRelevantLines). A list with all stanzas.
+    """
+    f = python_utils.open_file(LCOV_FILE_PATH, 'r')
+    lcov_items_list = f.read().split('end_of_record')
+    stanzas_list = []
+
+    for item in lcov_items_list:
+        relevant_lines = [line for line in item.splitlines() if
+                          any(line.startswith(prefix) for prefix in
+                          RELEVANT_LCOV_LINE_PREFIXES)]
+
+        if len(relevant_lines) > 0:
+            stanza = LcovStanzaRelevantLines(relevant_lines)
+            stanzas_list.append(stanza)
+
+    return stanzas_list
+
+
+def check_fully_covered_filenames_list_is_sorted():
+    """Check if fully_covered_filenames list is in alphabetic order."""
+    if fully_covered_filenames != sorted(fully_covered_filenames):
+        python_utils.PRINT(
+            'The \033[1mfully_covered_filenames\033[0m list must be kept'
+            ' in alphabetic order.')
+        sys.exit(1)
+
+
+def check_coverage_changes():
+    """Checks if the whitelist for fully covered files needs to be changed by:
+    - New file insertion
+    - File renaming
+    - File deletion
+
+    Raises:
+        Exception: LCOV_FILE_PATH doesn't exist.
+    """
+    if not os.path.exists(LCOV_FILE_PATH):
+        raise Exception('File at path {} doesn\'t exist'.format(
```

Comment:
Add a bit more context: Expected lcov file to be available at {}, but the file does not exist.

Reply:
Done